Generate AOTI input check by default

yushangdi opened the pull request:
Summary:
Generate AOTI size and stride input check by default. But the checks are only run if `AOT_INDUCTOR_DEBUG_COMPILE` env variable is set (to avoid slowing down the performance).

Example output:

```cpp
            bool _check_aoti_runtime_check_inputs_env() {
                const static char* env_var_value = getenv("AOTI_RUNTIME_CHECK_INPUTS");
                const static bool result = env_var_value != nullptr && env_var_value[0] != '\0';
                return result;
            }

            AOTI_NOINLINE static void __check_inputs_outputs(
                AtenTensorHandle* input_handles,
                AtenTensorHandle* output_handles) {
                if (!_check_aoti_runtime_check_inputs_env()){
                    return;
                }
//rest of the check
}

```

Test Plan: CI

Differential Revision: D70260490




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov @desertfire

desertfire commented on torch/_inductor/codegen/cpp_wrapper_cpu.py:
```diff
@@ -391,9 +391,20 @@ def gen_check(handle_kind, idx, name, tensor):
         self.prefix.splice(
             """
+            bool _check_aoti_runtime_check_inputs_env() {
+                static bool result = [] {
```
Do we have to use a lambda here?

desertfire replied:
I think as long as you declare `result` as static, its initialization will only be done once.

yushangdi replied:
@desertfire Changed to not using lambda now.

yushangdi committed:
Generate AOTI input check by default (#148005)

Summary:

Generate AOTI size and stride input check by default. But the checks are only run if `AOT_INDUCTOR_DEBUG_COMPILE` env variable is set (to avoid slowing down the performance).

Example output:

```cpp

            bool _check_aoti_runtime_check_inputs_env() {
                static bool result = [] {
                    const char* env_var_value = getenv("AOTI_RUNTIME_CHECK_INPUTS");
                    return env_var_value != nullptr && env_var_value[0] != '\0';
                }();
                return result;
            }

AOTI_NOINLINE static void __check_inputs_outputs(
                AtenTensorHandle* input_handles,
                AtenTensorHandle* output_handles) {
                if (!_check_aoti_runtime_check_inputs_env()){
                    return;
                }
//rest of the check
}

```

Test Plan:
```
buck run fbcode//mode/dev-nosan //caffe2/test/inductor:test_aot_inductor -- -r runtime_checks
buck run fbcode//mode/dev-nosan //caffe2/test/inductor:aot_inductor_arrayref_cpu -- -r test_runtime_checks_cpu_with_stack_allocation

```

Reviewed By: hl475

Differential Revision: D70260490
